Shared/Services/EthereumService.cs

```diff
@@ -34,7 +34,14 @@ public class EthereumService
                 var function = contract.GetFunction(functionName);
                 object[] parameters = new object[1] { tokenId! };
                 var uri = await function.CallAsync<string>(parameters);
-                return uri?.Remove(0, 7); //remove the ipfs portion
+                if (!String.IsNullOrEmpty(uri) && uri.Contains("ipfs://"))
+                {
+                    return uri?.Remove(0, 7);
+                }
+                else
+                {
+                    return uri;
+                }
             }
```

Comment:
This can be reduced to:

`return uri?.StartsWith("ipfs://") ?? false ? uri?.Remove(0, 7) : uri; //remove the ipfs portion`

Reply:
ah thats simpler. modified to suit